addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/ClientScriptBasedAuthenticationMethodType.java

```diff
@@ -351,6 +353,16 @@ private boolean hasBrowserLaunch(ZestScript zestScript) {
             return zestScript.getStatements().stream().anyMatch(ZestClientLaunch.class::isInstance);
         }
 
+        private void removeCloseStatements(ZestScript zestScript) {
+            for (int i = 0; i < zestScript.getStatements().size(); i++) {
+                ZestStatement stmt = zestScript.getStatements().get(i);
+                if (stmt instanceof ZestClientWindowClose close) {
+                    zestScript.getStatements().remove(i);
+                    i -= 1;
```

Comment:
`i--`?

Reply:
I'd suggest change that after (and everywhere, there are other places with the same).